### ISSUE TITLE
cop request doesn't retry when exceeding the deadline with tikv_client_read_timeout too small

### ISSUE THREAD

**Issue by crazycs520** [pingcap/tidb#48980]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
set @@tikv_client_read_timeout=1;
select count(*) from t1; -- execute in a loop.
```

### 2. What did you expect to see? (Required)

Execute success. TiDB should handle the `exceeding the deadline` error and auto-retry.

### 3. What did you see instead (Required)

Execute failed.

```sql
other error: Coprocessor task terminated due to exceeding the deadline
```

Related TiDB log:

```log
[2023/11/28 20:57:51.358 +08:00] [WARN] [coprocessor.go:1415] ["other error"] [conn=1730150422] [session_alias=] [txnStartTS=445953152470482969] [regionID=4878] [bucketsVer=0] [latestBucketsVer=0] [rangeNums=1] [firstRangeStartKey="t\ufffd\u0000\u0000\u0000\u0000\u0000\u0000n_r\u000100000000\ufffd00000000\ufffd00000000\ufffd025207\u0000\u0000\ufffd\u00010000-000\ufffd1-0002-0\ufffd004\u0000\u0000\u0000\u0000\u0000\ufffd"] [lastRangeEndKey="t\ufffd\u0000\u0000\u0000\u0000\u0000\u0000n_r\u000100000000\ufffd00000000\ufffd00000000\ufffd025207\u0000\u0000\ufffd\u00010000-000\ufffd1-0002-0\ufffd005\u0000\u0000\u0000\u0000\u0000\ufffd"] [storeAddr=10.0.1.9:22160] [kv_read_timeout=1] [error="other error: Coprocessor task terminated due to exceeding the deadline"]
[2023/11/28 20:57:51.358 +08:00] [INFO] [conn.go:1109] ["command dispatched failed"] [conn=1730150422] [session_alias=] [connInfo="id:1730150422, addr:127.0.0.1:65082 status:10, collation:utf8mb4_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select pk, sk, ts, v, n from (             select pk, sk, ts, v, row_number() over (partition by pk, sk order by ts desc) as n             from t1 where pk = '000000000000000000000000025207' and sk like '0000-0001-0002-0004%' and ts >= 1701176211351539000 and ts < 1701176261351546000         ) as tt where n <= 8 order by sk asc"] [txn_mode=PESSIMISTIC] [timestamp=445953152470482969] [err="other error: Coprocessor task terminated due to exceeding the deadline\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).handleCopResponse\n\t/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1410\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).handleCopPagingResult\n\t/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1344\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).handleTaskOnce\n\t/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1277\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).handleTask\n\t/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1130\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).run\n\t/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:817\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"]
```

### 4. What is your TiDB version? (Required)

master: 651e77099bbef516417b1c836e128f3515358fba

**Comment by cfzjywxk** [comment]
It is necessary to integrate the coprocessor endpoint of tikv to handle timeout error branches and standardize the corresponding error return types first. Afterward, it can be examined whether to handle it only in the kv client's `ClientHepler` or in the coprocessor response processing of tidb, trying to match specific error types to avoid string comparisons as much as possible.

**Comment by cfzjywxk** [comment]
> Better to use [tikv/tikv#16155](https://github.com/tikv/tikv/pull/16155) to fix this issue.

@crazycs520 Please pick the improvement/fix to the release-7.5 branch
